packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -124,6 +132,10 @@ class TestCommand extends FlutterCommand {
 
     Directory workDir;
     if (files.isEmpty) {
+      if (argResults['watch']) {
+        throwToolExit(
+          'When using --watch, you must specify one or more test file(s) to run.');
```

Comment:
Presumably the user could manually specify every single file right? Might as well make that supported fully

Reply:
The philosophy here is to make the main use case easy, and what we just allow to be done, but with some difficulty.  A sensible alternative would be to restrict to a single test file.  I don't have any _strong_ opinions either way, but we are solving for taking a single - or a small number of - test(s) from red to green.

Reply:
I agree in principle, but there is an edge case where the behavior is painful enough that we may as well allow it: the project as one or few tests files

Reply:
Agreed offline to remove this stricture